Documentation/rfc/v3api.proto

```diff
@@ -1,290 +0,0 @@
-  // A delete request increase the revision of the store,
-  // and generates one event in the event history.
-  rpc DeleteRange(DeleteRangeRequest) returns (DeleteRangeResponse) {}
-
-  // Txn processes all the requests in one transaction.
-  // A txn request increases the revision of the store,
-  // and generates events with the same revision in the event history.
-  // It is not allowed to modify the same key several times within one txn.
-  rpc Txn(TxnRequest) returns (TxnResponse) {}
-
-  // Compact compacts the event history in etcd. User should compact the
-  // event history periodically, or it will grow infinitely.
-  rpc Compact(CompactionRequest) returns (CompactionResponse) {}
-}
-
-service watch {
-  // Watch watches the events happening or happened. Both input and output
-  // are stream. One watch rpc can watch for multiple keys or prefixs and
-  // get a stream of events. The whole events history can be watched unless
-  // compacted.
-  rpc Watch(stream WatchRequest) returns (stream WatchResponse) {}
-}
-
-service Lease {
-  // LeaseCreate creates a lease. A lease has a TTL. The lease will expire if the
-  // server does not receive a keepAlive within TTL from the lease holder.
-  // All keys attached to the lease will be expired and deleted if the lease expires.
-  // The key expiration generates an event in event history.
-  rpc LeaseCreate(LeaseCreateRequest) returns (LeaseCreateResponse) {}
-
-  // LeaseRevoke revokes a lease. All the key attached to the lease will be expired and deleted.
-  rpc LeaseRevoke(LeaseRevokeRequest) returns (LeaseRevokeResponse) {}
-
-  // KeepAlive keeps the lease alive.
-  rpc LeaseKeepAlive(stream LeaseKeepAliveRequest) returns (stream LeaseKeepAliveResponse) {}
-
-  // TODO(xiangli) List all existing Leases?
-  // TODO(xiangli) Get details information (expirations, leased keys, etc.) of a lease?
-}
-
-message ResponseHeader {
-  // an error type message?
-  string error = 1;
-  uint64 cluster_id = 2;
-  uint64 member_id = 3;
-  // revision of the store when the request was applied.
-  int64 revision = 4;
-  // term of raft when the request was applied.
-  uint64 raft_term = 5;
-}
-
-message RangeRequest {
-  // if the range_end is not given, the request returns the key.
-  bytes key = 1;
-  // if the range_end is given, it gets the keys in range [key, range_end).
-  bytes range_end = 2;
-  // limit the number of keys returned.
-  int64 limit = 3;
-  // range over the store at the given revision.
-  // if revision is less or equal to zero, range over the newest store.
-  // if the revision has been compacted, ErrCompaction will be returned in
-  // response.
-  int64 revision = 4;
-}
-
-message RangeResponse {
-  ResponseHeader header = 1;
-  repeated storagepb.KeyValue kvs = 2;
-  // more indicates if there are more keys to return in the requested range.
-  bool more = 3;
-}
-
-message PutRequest {
-  bytes key = 1;
-  bytes value = 2;
-  int64 lease = 3;
```

Comment:
https://github.com/coreos/etcd/blob/master/etcdserver/etcdserverpb/rpc.proto#L90 is missing the lease field